### PR TITLE
Bump version for org.eclipse.m2e.maven.indexer

### PR DESCRIPTION
**Pull request by mickaelistria**
As there is some API change, fix version and requirements.

Signed-off-by: Mickael Istria <mistria@redhat.com>

### CODE REVIEW

File: m2e-maven-runtime/org.eclipse.m2e.maven.indexer/pom.xml [hunk 1]
```diff
@@ -19,6 +19,7 @@
   </parent>
 
   <artifactId>org.eclipse.m2e.maven.indexer</artifactId>
+  <version>1.17.1-SNAPSHOT</version>
```
Comment (fbricon) [inline m2e-maven-runtime/org.eclipse.m2e.maven.indexer/pom.xml:22]: 1.17.0 has not been released, why is this necessary?

File: org.eclipse.m2e.core.ui/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -28,7 +28,7 @@ Require-Bundle: org.eclipse.m2e.core;bundle-version="[1.16.0,2.0.0)",
  org.eclipse.m2e.model.edit;bundle-version="[1.16.0,2.0.0)";visibility:=reexport,
  org.eclipse.m2e.maven.runtime;bundle-version="[1.16.0,2.0.0)",
  org.eclipse.m2e.archetype.common;bundle-version="[1.16.0,2.0.0)",
- org.eclipse.m2e.maven.indexer;bundle-version="[1.16.0,2.0.0)",
+ org.eclipse.m2e.maven.indexer;bundle-version="[1.16.0.20201124-0221,2.0.0)",
```
Comment (fbricon) [inline org.eclipse.m2e.core.ui/META-INF/MANIFEST.MF:31]: [1.17.0,2.0.0)

**Review by fbricon — CHANGES_REQUESTED** [review]

File: m2e-maven-runtime/org.eclipse.m2e.maven.indexer/pom.xml (hunk 1 above)
Comment (mickaelistria) [reply m2e-maven-runtime/org.eclipse.m2e.maven.indexer/pom.xml:22]: 1.17.0 has been tagged and published, and then, I found this issue to fix...

**Comment by mickaelistria** [comment]
@mbooth101 FYI, this is an important change we forgot about when upgrading the indexer version. I would expect the build to detect such issues, but apparently the "compare-version-with-baselines" mojo isn't executing properly here to fail the build in such erroneous case.

**Comment by mbooth101** [comment]
@mickaelistria Sorry about this omission :-(

**Comment by mickaelistria** [comment]
@mbooth101 No big deal, it's not only yours and it actually reveals a build infra problem, so that's not all bad!
